src/python.cpp

```diff
@@ -1167,7 +1167,8 @@ PyObject* r_to_py_cpp(RObject x, bool convert) {
   // bytes
   } else if (type == RAWSXP) {
 
-    return as_python_bytes(RAW(sexp), Rf_length(sexp));
+    Rcpp::RawVector raw(sexp);
+    return PyByteArray_FromStringAndSize((const char*)&raw[0], raw.size());
```

Comment:
This is pedantic but `&raw[0]` could technically be an out-of-bounds access for an empty vector.

Reply:
Not pedantic at all as that's a real failure mode in both directions! (thanks for catching that). Just pushed a change, could you take a quick look (wrote tests but another pair of eyes always useful)